### PR TITLE
feat: Enhance Admin App Edit Keys Modal

### PR DESCRIPTION
**Pull request by CodeShakingSheep**
## What does this PR do?

I had a client complaint about the usability of the Edits Keys modal (especially for Jitsi and Nextcloud Talk apps). This PR attempts to improve overall user experience while adding an admin app by addresses the following issues:

- **Fix Dialog Stickiness**
The Dialog footer was sticky leading to it hiding some fields when there was a small height (e.g. Google Meet modal in the screenshots).
_Before_
image
_After_
image

- **Use password fields for sensitive data**
For all field names including `secret`, `password` or `key` a PasswordField is used now instead of a plain TextField.
- **Allow showing hints for fields**
For some fields a helper text is useful, especially for Jitsi and Nextcloud Talk path patterns. Therefore, I used Zod's description feature.
- **Allow showing a docs URL for integrating an app**
If there is a `docsUrl` defined in the app's config a link to it is shown in the bottom of the Edit Keys Modal.

**Jitsi Meet**
_Before_
image

_After_
image

**Nextcloud Talk**
_Before_
image

_After_
image

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox. N/A
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Navigate to admin apps, e.g. `https://localhost:3000/settings/admin/apps/conferencing` and click on the edit buttons (the buttons with the pencil) for Nextcloud Talk and Jitsi and see the above mentioned changes. Optionally, also click on the edit button of another app to see that there are no undesired changes (except the Password fields for sensitive data).

## Notes

This PR provides the base so that In the future for other apps hints and / or doc URLs could be added too.

To complete the PR it would probably be necessary to translate the added hints (for Jitsi and Nextcloud Talk apps), right? I just wanted to get a general approval for the PR before I dive into translations.

### CODE REVIEW

File: packages/features/apps/AdminAppsList.tsx [hunk 1]
```diff
@@ -232,22 +235,43 @@ const EditKeysModal: FC<{
...
-        <DialogFooter showDivider className="mt-8">
+        {!!docsUrl && (
+          <div className="text-subtle mb-3 w-full text-sm">
+            <a target="_blank" rel="noreferrer" className="hover:underline" href={docsUrl}>
+              <Icon name="book-open" className="-mt-1 mr-1 inline h-3 w-3" />
+              {t("documentation")}
+            </a>
+          </div>
+        )}
+        <DialogFooter noSticky showDivider className="mt-4">
```
Comment (anikdhabal) [inline packages/features/apps/AdminAppsList.tsx:274]: Hey, this dialog is not opened randomly. An admin will open this dialog only for setting the app credentials. We don't need to attach any documentation within the dialog; that's why we provide a README file to retrieve the credentials and set them accordingly

File: packages/app-store/nextcloudtalk/zod.ts [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-  nextcloudTalkClientSecret: z.string(),
+  nextcloudTalkHost: z.string().describe("Your Nextcloud base URL"),
+  nextcloudTalkPattern: z
+    .string()
+    .optional()
+    .describe(
+      "Pattern according to which your created calls are named. You can use the following placeholders: {uuid}, {Title}, {Event Type Title}, {Scheduler}, {Organizer}, {Location} and {Team}"
+    ),
+  nextcloudTalkClientId: z.string().describe("Your Nextcloud OAuth Client's ID"),
+  nextcloudTalkClientSecret: z.string().describe("Your Nextcloud OAuth Client's secret"),
```
Comment (anikdhabal) [inline packages/app-store/nextcloudtalk/zod.ts:12]: I think it's understandable that it is the clientId and clientSecret fields from the props name

File: packages/features/apps/AdminAppsList.tsx [hunk 3]
```diff
@@ -232,22 +235,43 @@ const EditKeysModal: FC<{
...
+                      hint={
+                        (appKeySchema.shape as Record<string, { description?: string } | undefined>)[key]
+                          ?.description
+                      }
+                      onChange={(e) => {
+                        formMethods.setValue(key, e?.target.value);
+                      }}
+                    />
+                  );
+                }}
```
Comment (anikdhabal) [inline packages/features/apps/AdminAppsList.tsx:261]: Hey, as I mentioned, only an admin can open this modal and set values, so I don't think any changes are needed here

**Review by anikdhabal — CHANGES_REQUESTED** [review]
Left comments

File: packages/features/apps/AdminAppsList.tsx (hunk 1 above)
Comment (CodeShakingSheep) [reply packages/features/apps/AdminAppsList.tsx:274]: Hey,
Thank you for reviewing. I understand. However, the README file is not even available in the UI. So, I believe it's a bit cumbersome to access for a non-developer. I'd be open to another solution though (e.g. making the README file available in the UI). Adding a link to a docs URL was just the easiest option to implement for me. What do you think?

File: packages/app-store/nextcloudtalk/zod.ts (hunk 2 above)
Comment (CodeShakingSheep) [reply packages/app-store/nextcloudtalk/zod.ts]: Oh yes, agreed. I just still had them in there for testing. Removed them.

File: packages/features/apps/AdminAppsList.tsx (hunk 3 above)
Comment (CodeShakingSheep) [reply packages/features/apps/AdminAppsList.tsx:261]: Which changes are you referring to exactly?

Using password fields for sensitive data is a general measurement to increase data security. No matter if admin or not there's always the possibility that someone's looking at their screen.

Adding the hints was especially necessary for the Jitsi and Nextcloud Talk path patterns. Jitsi doesn't even have a README file. When I was using the Jitsi app in Cal.com for the first time I had to look into the code to understand how that field is working. I think that's not feasible for a non-dev admin. So, I'd rather keep those hints. Or do you have another suggestion to document the usage of those fields?

Comment (anikdhabal) [reply packages/features/apps/AdminAppsList.tsx:261]: No, I don't think attaching a documentation URL could resolve user confusion. There isn’t a specific URL that provides a step-by-step guide for retrieving credentials and similar tasks. That's why we always ask for a well-written README. Could you pls add A Readme for Jitsi

**Comment by anikdhabal** [comment]
Closing it due to staleness

**Pull request closed without merging**